apps/web/pages/event-types/[type].tsx

```diff
@@ -1081,6 +1086,55 @@ const EventTypePage = (props: inferSSRProps<typeof getServerSideProps>) => {
                           )}
                         />
 
+                        <hr className="my-2 border-neutral-200" />
+
+                        {hasRainbowIntegration && (
+                          <>
+                            <div className="block items-center sm:flex">
+                              <div className="min-w-48 mb-4 sm:mb-0">
+                                <label
+                                  htmlFor="blockchainId"
+                                  className="flex text-sm font-medium text-neutral-700">
+                                  {t("Blockchain")}
+                                </label>
+                              </div>
+                              <Select
+                                isSearchable={false}
+                                className="block w-full min-w-0 flex-1 rounded-sm text-sm"
+                                onChange={(val) => {
+                                  formMethods.setValue("blockchainId", val || 1);
+                                }}
+                                defaultValue={
+                                  SUPPORTED_CHAINS_FOR_FORM.find(
+                                    (e) => e.value === eventType.metadata.blockchainId
+                                  ) || { value: 1, label: "Ethereum" }
+                                }
+                                options={SUPPORTED_CHAINS_FOR_FORM || [{ value: 1, label: "Ethereum" }]}
+                              />
+                            </div>
+                            <div className="block items-center sm:flex">
+                              <div className="min-w-48 mb-4 sm:mb-0">
+                                <label
+                                  htmlFor="smartContractAddress"
+                                  className="flex text-sm font-medium text-neutral-700">
```

Comment:
same as here, we don't want to hard code web3 stuff into /[type].tsx

Reply:
extracted here https://github.com/calcom/cal.com/pull/4019/commits/dd47071d4f5fd693b2dcb77ef7d96c6de3ee4287